test/distributed/test_c10d_common.py

```diff
@@ -1116,6 +1116,26 @@ def _test_warn_not_in_group(self, backend):
             dist.barrier(group=group)
             dist.broadcast(x, src=0, group=group)
 
+    def _test_new_sub_group(self, backend):
+        store = dist.FileStore(self.file_name, self.world_size)
+        dist.init_process_group(
+            backend,
+            world_size=self.world_size,
+            rank=self.rank,
+            store=store,
+        )
+
+        rank = dist.get_rank()
+        ranks_in = [rank]
+        ranks_out = [(rank + 1) % self.world_size]
+        self.assertIn(rank, ranks_in)
+        self.assertNotIn(rank, ranks_out)
+
+        with self.assertRaisesRegex(RuntimeError, "new_sub_group can only be called from a member rank"):
+            dist.new_sub_group(ranks=ranks_out)
+
+        self.assertIsInstance(dist.new_sub_group(ranks=ranks_in), dist.ProcessGroup)
```

Comment:
should we also add testing where we use `new_sub_group` with > 1 rank and where we call it on some ranks but not the others?

Reply:
good point, it requires some larger changes to the test suite but makes sense to have it.